schlib/rules/S6_2.py

```diff
@@ -93,12 +93,69 @@ def checkDatasheet(self):
 
         # Datasheet field must be empty
         if not self.checkEmpty(ds):
-            self.error("Datasheet field must be EMPTY")
+            self.error("Datasheet field (.lib file) must be EMPTY")
             fail = True
 
         return fail
 
-    def check(self):
+    def checkDocumentation(self, name, documentation, alias=False, isGraphicOrPowerSymbol=False):
+
+        errors = []
+        warnings = []
+
+        if not documentation:
+            errors.append("Missing all metadata information in the .dcm file (datasheet, keyword and description)")
+        elif (not documentation['description'] or
+            not documentation['keywords'] or
+            not documentation['datasheet']):
+
+            if (not documentation['description']):
+                errors.append("Missing DESCRIPTION entry (in dcm file)")
+            if (not documentation['keywords']):
+                errors.append("Missing KEYWORDS entry (in dcm file)")
+            if (not isGraphicOrPowerSymbol) and (not documentation['datasheet']):
+                errors.append("Missing DATASHEET entry (in dcm file)")
+
+                if (documentation['description'] and
+                    documentation['keywords']):
+                    self.only_datasheet_missing = True
+
+        # Symbol name should not appear in the description
+        desc = documentation.get('description', '')
+        if desc and name.lower() in desc.lower():
+            warnings.append("Symbol name should not be included in description")
+
+        # Datasheet field should look like a a datasheet
+        ds = documentation.get('datasheet', '')
+
+        if ds and len(ds) > 2:
+            link = False
+            links = ['http', 'www', 'ftp']
+            if any([ds.startswith(i) for i in links]):
```

Comment:
Perhaps this could be more robust. I'm sure there are good regex for a URL online. Some things I can think of now:
1. `www2` instead of just `www` or `https` instead of just `http`. This will currently work because the starting characters are the same, but it's so rudimentary.
1. Check for `://` or other URL formatting.
1. `html` instead of just `htm`.

And there could be a warning if the link is to a webpage instead of a PDF file. Sometimes that's the best way but the vast majority of our links are PDFs.

Reply:
How about actually fetching the URL?
* Throw an error if the final http status code is not `200 OK`
* Check filetype (perhaps with `python-magic`)

Reply:
Not sure if having the script fetch and process whatever URL random people on the internet throw at it is a good idea.

Reply:
Can be done in a later pull request. This request will only combine the checks of two rules and align the error messages with the current kicad gui